plugins/inputs/bigbluebutton/api.go

```diff
@@ -0,0 +1,51 @@
+package bigbluebutton
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const getMeetingCallName = "getMeetings"
+const getRecordingsCallName = "getRecordings"
+const NoRecordingMessageKey = "noRecordings"
+const NoMeetingsMessageKey = "noMeetings"
+
+// BigBlueButton uses an authentication based on a SHA1 checksum processed from api call name and server secret key
+func (b *BigBlueButton) checksum(apiCallName string) []byte {
+	hash := sha1.New()
+	hash.Write([]byte(fmt.Sprintf("%s%s", apiCallName, b.SecretKey)))
+	return hash.Sum(nil)
+}
+
+// Call BBB server api
+func (b *BigBlueButton) api(apiCallName string) ([]byte, error) {
+	endpoint := fmt.Sprintf("%s%s", b.APIEndpoint, apiCallName)
+
+	url := fmt.Sprintf("%s%s?checksum=%x", b.URL, endpoint, b.checksum(apiCallName))
```

Comment:
To be honest I like to see this in an `Init()` function to precompute the URL in order to avoid the checksum calculation in each gather cycle. This will save a few cycles...